### DISCUSSION TITLE
LILYGO T-Dongle-S3 Displaydesc ST3577

### DISCUSSION THREAD

**Discussion by SN-8023** [arendst/Tasmota#19781] [Q&A]
```
EPS32-S3, 16MB, 0.96" ISP LCD, 80x160
TFT_RESET   GPIO 01 -> display Reset
TFT_DC      GPIO 02 -> SSPI DC
TFT_SDA     GPIO 03 -> SSPI MOSI
TFT_CS      GPIO 04 -> SSPI_CS
TFT_SCL     GPIO 05 -> SSPI_CLK
TFT_LED     GPIO 38 -> Backlight

SDMMC_D0    GPIO 14
SDMMC_D1    GPIO 17
SDMMC_D2    GPIO 21
SDMMC_D3    GPIO 18
SDMMC_CLK   GPIO 12
SDMMC_CMD   GPIO 16

APA102 RGB_LED
LED_CLK     GPIO39
LED_DI      GPIO40
```

I still have problems with the displaydesc...
I have managed so far that everything is displayed where it belongs.
Only with the colors it does not work. Black is white, red is yellow, etc. and I don't really understand the concept of :R, :0, :1, :2 and :3 in relation to the colors.
Where do I have to do something to make something happen? 🤔

```
:H,Lilygo_T-Dongle-S3,80,160,16,SPI,3,4,5,3,2,38,1,-1,20
:S,2,1,1,0,20,20
:I
01,80
11,80
B1,3,01,2C,2D
B2,3,01,2C,2D
B3,6,01,2C,2D,01,2C,2D
B4,1,07
C0,3,A2,04,84
C1,1,C5
C2,2,0A,00
C3,2,8A,2A
C4,2,8A,EE
C5,1,0E
20,0
36,1,C8
3A,1,05
2A,4,00,02,00,7F
2B,4,00,01,00,7F
E0,10,02,1C,07,12,37,32,29,2D,29,25,2B,39,00,01,03,10
E1,10,03,1D,07,06,2E,2C,29,2D,2E,2E,37,3F,00,00,02,10
13,80
29,80
:o,28
:O,29
:A,2A,2B,2C,16
:R,36
:0,c8,1a,01,00
:1,A0,01,1a,01
:2,00,1a,01,02
:3,60,01,1a,03
:B,60,1
#
```

**Comment by ToeiRei** [comment]
Not sure about tasmota here, but to my knowledge, the T-Dongle S3 uses inverted colors.

**Comment by dexif** [comment]
Did you manage to set up your device?

**Reply by SN-8023** [reply]
no sorry, the colors are still weird... but I haven't looked into it any further. I only bought the dongle to play or try it out.

**Reply by dexif** [reply]
I couldn't get the screen to work at all. Only the backlight lights up and that’s it.

**Reply by SN-8023** [reply]
I checked again, the Display.ini is still the same as above. Do you have the LiliGo dongle or just the display?

with
```
DisplayRows 10
DisplayCols 26
DisplayRotate 1
DisplayMode 2
```
is nothing coming?

**Reply by dexif** [reply]
Yes, it's a dongle.I flashed the latest firmware. I tried and s3-lvgl and just lvgl. Everything works except the display :( I think the problem is in the latest firmware. Or I didn't understand correctly how it works. Perhaps I needed to compile the firmware specifically for this dongle?

**Reply by sfromis** [reply]
With latest development branch firmware, you *have to* use the [Universal Display Driver](https://tasmota.github.io/docs/Universal-Display-Driver/), meaning no compile depending on display type, as long as you set up both the template (including Option A 3) and `display.ini` to match. This will be the standard for the upcoming Tasmota 14.0 release and forward.

Note: The info from Lilygo does not tell what ESP32-S3 chip variant it is, but does specify 16 MB of flash (which is fine), without any mention of PSRAM. While you can get basic stuff working on the display without PSRAM, you may get into trouble if you want to have fun with images and fonts. Of course, the tiny display does not lend itself to going crazy there, but still...

**Reply by SN-8023** [reply]
the silkscreen at least suggests that psram is present. is there a way to read it out?
![IMG_2605](https://github.com/arendst/Tasmota/assets/65110735/2af6e206-b4b4-4ec5-858c-64d1c6cd9449)

**Reply by sfromis** [reply]
If Tasmota detects PSRAM present, it will report the available size in `PS-RAM Memory` near the bottom of the "Information" page in the web UI, and generally also something like this as first message after boot:
`00:00:00.001 HDW: ESP32-S3 v0.2 (PSRAM)`

On the chip itself (not really readable in your image), there should also be a nomenclature **if** the flash and PSRAM is on-chip, with the size after the letter "R" somewhere. Your board looks like having a separate flash chip, which then is not in the chip nomenclature.
![image](https://github.com/arendst/Tasmota/assets/47082390/a40f0ac3-a5ac-43f5-be07-ab1248237b94)

**Reply by SN-8023** [reply]
then it probably has none... Too bad.

```
Program Version	13.1.0(tasmota32)
Build Date & Time	2023-10-17T22:05:49
Core/SDK Version	2_0_11/4.4.5.230722
 
ESP Chip Id	5822952 (ESP32-S3)
Flash Chip Id	0x1840EF (QIO)
Flash Size	16384 KB
Program Flash Size	16384 KB
Program Size	1837 KB
Free Program Space	1042 KB
Free Memory	221.8 KB (frag. 7%)
 
Partition safeboot	832 KB (used 87%)
Partition app0*	2880 KB (used 63%)
Partition fs	12608 KB
```

**Reply by sfromis** [reply]
A reason for me talking about the nomenclature on the chip itself is that some variants are using octal PSRAM, needing a different build. If not using that, PSRAM cannot be detected at boot. Anyway, as the dongle is not a recent device, I'd not really expect it to have octal PSRAM. Should still be viable for basic usage.

**Reply by SN-8023** [reply]
![IMG_2606](https://github.com/arendst/Tasmota/assets/65110735/b4dd0f0e-749b-4836-9e92-92a81bdd2ad1)

it's a nice little dongle and for the size a lot of features, RGB LED (even if it's not a WS281x), TransFlash slot, display, wifi/bt...

the case could be a bit more premium for my taste, but it is what it is...
